From 10e3b5858588d81776a4220dc161a388e9a76652 Mon Sep 17 00:00:00 2001
From: Bharat B Panda <bharat.b.panda@intel.com>
Date: Fri, 21 Jul 2023 13:20:57 +0530
Subject: [PATCH] Add PCI device configuration for stmmac

Assign PCI dev id and load respecting PCI configuration for stmmac.

Signed-off-by: Bharat B Panda <bharat.b.panda@intel.com>
---
 drivers/net/ethernet/stmicro/stmmac/dwmac-intel.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/net/ethernet/stmicro/stmmac/dwmac-intel.c b/drivers/net/ethernet/stmicro/stmmac/dwmac-intel.c
index 5a22046cf12c..8cbbdb4219c3 100644
--- a/drivers/net/ethernet/stmicro/stmmac/dwmac-intel.c
+++ b/drivers/net/ethernet/stmicro/stmmac/dwmac-intel.c
@@ -83,6 +83,7 @@ static int intel_serdes_powerup(struct net_device *ndev, void *priv_data)
 	int serdes_phy_addr = 0;
 	u32 data = 0;
 
+	printk("intel_serdes_powerup \n");
 	if (!intel_priv->mdio_adhoc_addr)
 		return 0;
 
@@ -908,6 +909,7 @@ static int tgl_common_data(struct pci_dev *pdev,
 static int tgl_sgmii_phy0_data(struct pci_dev *pdev,
 			       struct plat_stmmacenet_data *plat)
 {
+	printk("tgl_sgmii_phy0_data: platform detected, config values");
 	plat->bus_id = 1;
 	plat->dma_cfg->pch_intr_wa = 1;
 	plat->phy_interface = PHY_INTERFACE_MODE_SGMII;
@@ -992,6 +994,7 @@ static struct stmmac_pci_info adls_sgmii1g_phy1_info = {
 static int adln_sgmii_phy0_data(struct pci_dev *pdev,
 				struct plat_stmmacenet_data *plat)
 {
+	printk("adln_sgmii_phy0_data: platform detected, config values");
 	plat->bus_id = 1;
 	plat->phy_interface = PHY_INTERFACE_MODE_SGMII;
 	plat->serdes_powerup = intel_serdes_powerup;
@@ -1482,6 +1485,7 @@ static const struct dev_pm_ops intel_eth_pm_ops = {
 #define PCI_DEVICE_ID_INTEL_ADLS_SGMII1G_0	0x7aac
 #define PCI_DEVICE_ID_INTEL_ADLS_SGMII1G_1	0x7aad
 #define PCI_DEVICE_ID_INTEL_ADLN_SGMII1G	0x54ac
+#define PCI_DEVICE_ID_INTEL_RPLP_SGMII1G	0x51ac
 
 static const struct pci_device_id intel_eth_pci_id_table[] = {
 	{ PCI_DEVICE_DATA(INTEL, QUARK, &quark_info) },
@@ -1500,6 +1504,7 @@ static const struct pci_device_id intel_eth_pci_id_table[] = {
 	{ PCI_DEVICE_DATA(INTEL, ADLS_SGMII1G_0, &adls_sgmii1g_phy0_info) },
 	{ PCI_DEVICE_DATA(INTEL, ADLS_SGMII1G_1, &adls_sgmii1g_phy1_info) },
 	{ PCI_DEVICE_DATA(INTEL, ADLN_SGMII1G, &adln_sgmii1g_phy0_info) },
+	{ PCI_DEVICE_DATA(INTEL, RPLP_SGMII1G, &tgl_sgmii1g_phy0_info) },
 	{}
 };
 MODULE_DEVICE_TABLE(pci, intel_eth_pci_id_table);
-- 
2.40.1

